diff --git a/modules/cohesion_elements/src/Plugin/Field/FieldWidget/CohesionLayoutBuilderWidget.php b/modules/cohesion_elements/src/Plugin/Field/FieldWidget/CohesionLayoutBuilderWidget.php
index e7f282d..bcc8da0 100644
--- a/modules/cohesion_elements/src/Plugin/Field/FieldWidget/CohesionLayoutBuilderWidget.php
+++ b/modules/cohesion_elements/src/Plugin/Field/FieldWidget/CohesionLayoutBuilderWidget.php
@@ -93,20 +93,18 @@ class CohesionLayoutBuilderWidget extends WidgetBase implements ContainerFactory
     $values = $items->getValue();
     $target_type = $this->getFieldSetting('target_type');
     $entity_storage = $this->entityTypeManager->getStorage($target_type);
+    // If the form loads after node preview, there is already an entity
+    // attached with potentially modified values. Use that if possible, so we
+    // don't lose work in progress.
+    if (isset($values[$delta]['entity']) && $values[$delta]['entity'] instanceof CohesionLayout) {
+      $layout_entity = $values[$delta]['entity'];
+    }
     /** @var \Drupal\cohesion_elements\Entity\CohesionLayout $layout_entity */
-    if (empty($values[$delta]['target_id'])) {
+    elseif (empty($values[$delta]['target_id'])) {
       $layout_entity = $entity_storage->create();
     }
     else {
-      // If the form loads after node preview, there is already an entity
-      // attached with potentially modified values. Use that if possible, so we
-      // don't lose work in progress.
-      if (isset($values[$delta]['entity']) && $values[$delta]['entity'] instanceof CohesionLayout) {
-        $layout_entity = $values[$delta]['entity'];
-      }
-      else {
-        $layout_entity = $entity_storage->loadRevision($values[$delta]['target_revision_id']);
-      }
+      $layout_entity = $entity_storage->loadRevision($values[$delta]['target_revision_id']);
     }
 
     $storage = $form_state->getStorage();
